src/backend/updates.rs

```diff
@@ -71,19 +78,93 @@ fn post_upgrade() {
 }
 
 async fn post_upgrade_fixtures() {
+    mutate(|state| state.read_only = true);
+    let total_posts = read(|state| state.posts.len());
+
+    // Init all stable structs becasue we do not need the old contents of the stable memory anymore
+    init_stable_structs();
+
+    loop {
+        let num_posts = read(|state| state.posts.len());
+        let _: Result<(), ()> = canisters::call_canister(id(), "migrate_posts", ((),))
+            .await
+            .map_err(|(code, err)| {
+                mutate(|state| {
+                    state.logger.error(format!(
+                        "couldn't migrate posts: code {:?}, err {}",
+                        code, err
+                    ))
+                })
+            });
+        let new_num_posts = read(|state| state.posts.len());
+        if new_num_posts == 0 {
+            // We're done with migration
+            break;
+        }
+        if num_posts == new_num_posts && num_posts > 0 {
+            // Something went wrong, so we stop and let the state locked.
+            mutate(|state| {
+                state
+                    .logger
+                    .error("couldn't migrate posts: something went wrong")
+            });
+            return;
+        }
+    }
+
+    if total_posts as u64 == Post::count() {
+        mutate(|state| {
+            state.read_only = false;
+            state
+                .logger
+                .debug(format!("Succefully migrated {} posts", total_posts))
+        })
+    }
+}
```

Comment:
Let's also introduce an endpoint that copies the body of the loop above. I.e. represents one step of migration. This would allow us to complete the migration manually even if the loop above stops for some reason.

Reply:
I was thinking about
```
#[update]
fn migration_step(num_posts_to_migrate) {
  // The body of the loop that call internally `migrate_posts(num_posts_to_migrate)` but also has the logic of moving failed posts to `state.failed_posts` and logic to complete migration if there are no posts remaining (copying `state.failed_posts` back to `state.posts`)
}
```